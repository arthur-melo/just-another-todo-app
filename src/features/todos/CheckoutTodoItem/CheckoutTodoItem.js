import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSquare, faCheckSquare } from '@fortawesome/free-solid-svg-icons';

const checkoutTodoItemPropTypes = {
  isCompleted: PropTypes.bool.isRequired,
};

const CheckoutTodoItem = props => {
  const { isCompleted } = props;

  return (
    <button type="button" className="border border-0 p-0 bg-transparent">
      {isCompleted ? (
        <FontAwesomeIcon icon={faCheckSquare} fontSize="36px" fixedWidth />
      ) : (
        <FontAwesomeIcon icon={faSquare} fontSize="36px" fixedWidth />
      )}
    </button>
  );
};

CheckoutTodoItem.propTypes = checkoutTodoItemPropTypes;

export default CheckoutTodoItem;
