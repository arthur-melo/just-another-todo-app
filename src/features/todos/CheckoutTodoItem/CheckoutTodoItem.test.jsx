import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import CheckoutTodoItem from './CheckoutTodoItem';

describe('CheckoutTodoItem', () => {
  it('should render the component with a checkmark icon when isCompleted prop is true', async () => {
    const completed = true;

    const { asFragment } = render(<CheckoutTodoItem isCompleted={completed} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the component with a square icon when isCompleted prop is false', async () => {
    const completed = false;

    const { asFragment } = render(<CheckoutTodoItem isCompleted={completed} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
