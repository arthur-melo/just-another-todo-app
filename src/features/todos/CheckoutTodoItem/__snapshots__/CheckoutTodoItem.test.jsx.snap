// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CheckoutTodoItem > should render the component with a checkmark icon when isCompleted prop is true 1`] = `
<DocumentFragment>
  <button
    class="border border-0 p-0 bg-transparent"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-square-check fa-fw "
      data-icon="square-check"
      data-prefix="fas"
      focusable="false"
      font-size="36px"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M64 32C28.7 32 0 60.7 0 96L0 416c0 35.3 28.7 64 64 64l320 0c35.3 0 64-28.7 64-64l0-320c0-35.3-28.7-64-64-64L64 32zM337 209L209 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L303 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
        fill="currentColor"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`CheckoutTodoItem > should render the component with a square icon when isCompleted prop is false 1`] = `
<DocumentFragment>
  <button
    class="border border-0 p-0 bg-transparent"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-square fa-fw "
      data-icon="square"
      data-prefix="fas"
      focusable="false"
      font-size="36px"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 96C0 60.7 28.7 32 64 32H384c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96z"
        fill="currentColor"
      />
    </svg>
  </button>
</DocumentFragment>
`;
